### PR TITLE
Tabular: Refactored model names

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
#782 

*Description of changes:*

- Changed model names based on agreed proposals in #782. 
- This PR does not incorporate the level+1 change due to the logic changes required (will do in separate PR).
- Keeps KNeighbors to avoid unnecessarily long name.
- NeuralNet -> NeuralNetMXNet

```
0                  CatBoost_BAG_L0
1                  CatBoost_BAG_L1
2             CatBoost_FULL_BAG_L0
3             CatBoost_FULL_BAG_L1
4            ExtraTreesEntr_BAG_L0
5            ExtraTreesEntr_BAG_L1
6       ExtraTreesEntr_FULL_BAG_L0
7       ExtraTreesEntr_FULL_BAG_L1
8            ExtraTreesGini_BAG_L0
9            ExtraTreesGini_BAG_L1
10      ExtraTreesGini_FULL_BAG_L0
11      ExtraTreesGini_FULL_BAG_L1
12           KNeighborsDist_BAG_L0
13           KNeighborsDist_BAG_L1
14      KNeighborsDist_FULL_BAG_L0
15      KNeighborsDist_FULL_BAG_L1
16           KNeighborsUnif_BAG_L0
17           KNeighborsUnif_BAG_L1
18      KNeighborsUnif_FULL_BAG_L0
19      KNeighborsUnif_FULL_BAG_L1
20           LightGBMCustom_BAG_L0
21           LightGBMCustom_BAG_L1
22      LightGBMCustom_FULL_BAG_L0
23      LightGBMCustom_FULL_BAG_L1
24               LightGBMXT_BAG_L0
25               LightGBMXT_BAG_L1
26          LightGBMXT_FULL_BAG_L0
27          LightGBMXT_FULL_BAG_L1
28                 LightGBM_BAG_L0
29                 LightGBM_BAG_L1
30            LightGBM_FULL_BAG_L0
31            LightGBM_FULL_BAG_L1
32           NeuralNetMXNet_BAG_L0
33           NeuralNetMXNet_BAG_L1
34      NeuralNetMXNet_FULL_BAG_L0
35      NeuralNetMXNet_FULL_BAG_L1
36         RandomForestEntr_BAG_L0
37         RandomForestEntr_BAG_L1
38    RandomForestEntr_FULL_BAG_L0
39    RandomForestEntr_FULL_BAG_L1
40         RandomForestGini_BAG_L0
41         RandomForestGini_BAG_L1
42    RandomForestGini_FULL_BAG_L0
43    RandomForestGini_FULL_BAG_L1
44        WeightedEnsemble_FULL_L1
45        WeightedEnsemble_FULL_L2
46             WeightedEnsemble_L1
47             WeightedEnsemble_L2
48                  XGBoost_BAG_L0
49                  XGBoost_BAG_L1
50             XGBoost_FULL_BAG_L0
51             XGBoost_FULL_BAG_L1
Name: model, dtype: object
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-800-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-800/1/index.html

File: core/src/autogluon/core/constants.py [hunk 1]
```diff
@@ -25,9 +25,8 @@
 """
 AG_ARGS: Dictionary of customization options related to meta properties of the model such as its name, the order it is trained, and the problem types it is valid for.
     name: (str) The name of the model. This overrides AutoGluon's naming logic and all other name arguments if present.
-    name_main: (str) The main name of the model. In 'RandomForestClassifier', this is 'RandomForest'.
+    name_main: (str) The main name of the model.
```
Comment (jwmueller) [inline core/src/autogluon/core/constants.py:28]: Maybe keep `Example: 'RandomForest'`

File: tabular/src/autogluon/tabular/task/tabular_prediction/tabular_prediction.py [hunk 2]
```diff
@@ -259,9 +259,8 @@ def fit(train_data,
                     AG_args: Dictionary of customization options related to meta properties of the model such as its name, the order it is trained, and the problem types it is valid for.
                         Valid keys:
                             name: (str) The name of the model. This overrides AutoGluon's naming logic and all other name arguments if present.
-                            name_main: (str) The main name of the model. In 'RandomForestClassifier', this is 'RandomForest'.
+                            name_main: (str) The main name of the model.
```
Comment (jwmueller) [inline tabular/src/autogluon/tabular/task/tabular_prediction/tabular_prediction.py:262]: Maybe keep `Example: 'RandomForest'`

**Review by jwmueller — APPROVED** [review]
looks good

File: core/src/autogluon/core/constants.py (hunk 1 above)
Comment (Innixma) [reply core/src/autogluon/core/constants.py:28]: Added

File: tabular/src/autogluon/tabular/task/tabular_prediction/tabular_prediction.py (hunk 2 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/task/tabular_prediction/tabular_prediction.py]: Added

**Comment by szha** [comment]
Job PR-800-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-800/2/index.html

**Pull request merged**